File: TeamProject.Models/Models/Booking.cs
using System.ComponentModel.DataAnnotations;

namespace TeamProject.Models.Models;

public class Booking
{
    [Key]
    public string Id { get; set; } = Guid.NewGuid().ToString();

    public DateTime CheckInDate { get; set; }

    public DateTime CheckOutDate { get; set; }

    public decimal BookingPrice { get; set; }

    public int PropertyId { get; set; }
    public Property Property { get; set; } = null!;

    public string CustomerUserId { get; set; } = string.Empty;
    public Customer Customer { get; set; } = null!;
}

File: src/lib/server/db/entities/booking.ts
import {
  Column,
  CreateDateColumn,
  Entity,
  JoinColumn,
  ManyToOne,
  OneToOne,
  PrimaryGeneratedColumn,
  type Relation,
} from "typeorm";
import { Customer } from "./customer";
import { Property } from "./property";
import { Review } from "./review";

@Entity({ name: "bookings" })
export class Booking {
  @PrimaryGeneratedColumn("uuid")
  id!: string;

  @CreateDateColumn()
  bookedAt!: Date;

  @Column({ type: "datetime" })
  checkInDate!: Date;

  @Column({ type: "datetime" })
  checkOutDate!: Date;

  @Column({ type: "decimal", precision: 18, scale: 2 })
  bookingPrice!: number;

  @Column({ type: "text" })
  propertyId!: string;

  @Column({ type: "text" })
  customerUserId!: string;

  // many bookings may belong to one property
  @ManyToOne(
    () => Property,
    (property) => property.bookings,
    { onDelete: "RESTRICT" },
  )
  @JoinColumn({ name: "propertyId", referencedColumnName: "id" })
  property!: Relation<Property>;

  // many bookings may belong to one customer
  @ManyToOne(
    () => Customer,
    (customer) => customer.bookings,
    { onDelete: "RESTRICT" },
  )
  @JoinColumn({ name: "customerUserId", referencedColumnName: "userId" })
  customer!: Relation<Customer>;

  // one booking may have one review
  @OneToOne(
    () => Review,
    (review) => review.booking,
  )
  review!: Relation<Review>;
}
